<launch>
  <!-- This node allows you to visually edit the regions for each component on top
       of a reference board image, and save the new CSV to be used by the evaluator. -->

  <!-- Manually step through frames with key presses -->
  <arg name='step' default='false' />
  <!-- Directory to find ROI csv files -->
  <arg name='dir' default='$(find nist_atb_eval)/config' />
  <!-- csv file to load -->
  <arg name='in_csv' default='tb_roi.csv' />
  <!-- csv file to save -->
  <arg name='out_csv' default='$(arg in_csv)' />
  <!-- Reference image to overlay ROIs onto -->
  <arg name='ref_img' default='$(find nist_atb_eval)/data/test/test01.png' />

  <!-- Start the node with given arguments -->
  <node name='roi_config' pkg='nist_atb_eval' type='roi_config.py' output='screen' required='true'>
    <param name='step' value='$(arg step)' />
    <param name='dir' value='$(arg dir)' />
    <param name='in_csv' value='$(arg in_csv)' />
    <param name='out_csv' value='$(arg out_csv)' />
    <param name='ref_image' value='$(arg ref_img)' />
  </node>
</launch>
